### PR TITLE
Save payment success appear when cancel #146837101

### PR DESCRIPTION
**Pull request by triitvn**

### CODE REVIEW

File: src/directives/pp-message/pp-message-container.directive.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        scope.$on('$destroy', function() {
+          element.off('click', eventHandler);
+        });
+
+        function eventHandler(event) {
+          // https://blog.lelonek.me/angularjs-and-scope-apply-best-practices-d07353175024
+          // Because it use element.on click event handler
+          scope.$apply(function() {
+            scope.$broadcast(ppMessageContants.BROADCAST_EVENT_NAME_DISMISS_MESSAGE, {
```
Comment (chigia001) [inline src/directives/pp-message/pp-message-container.directive.js]: As I understand, you listen on click event of root element. If you receive any click event you will hide all the event message container?

What happen if we click on different element like modal?

File: src/directives/pp-message/pp-message.directive.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+(function () {
+  'use strict';
+
+  angular
+    .module('pp_shared.directives.ppMessage')
+    .directive('ppMessage', ['ppMessageContants', function(ppMessageContants) {
+      return {
+        restrict: 'EA',
+        scope: {
+          message: '<',
```
Comment (chigia001) [inline src/directives/pp-message/pp-message.directive.js]: one-way binding is not working(as expected) for object.http://jslancer.com/2016/02/19/demystify-angularjs-15-one-way-data-binding/

Comment (chigia001) [inline src/directives/pp-message/pp-message.directive.js]: Make sure one-way binding is supported for angular 1.4

File: src/directives/pp-message/pp-message-container.directive.js (hunk 1 above)
Comment (triitvn) [inline src/directives/pp-message/pp-message-container.directive.js]: Fixed: bind event into document https://github.com/triitvn/pp-shared/pull/224/files#diff-a7a4eaa7e0c99aa5c1df81153c586bdcR6762

File: src/directives/pp-message/pp-message.directive.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          type: '@' // 'success' or 'error'
```
Comment (triitvn) [inline src/directives/pp-message/pp-message.directive.js]: Removed

Comment (triitvn) [inline src/directives/pp-message/pp-message.directive.js]: Updated https://github.com/triitvn/pp-shared/pull/224/files#diff-a7a4eaa7e0c99aa5c1df81153c586bdcR6803

File: src/directives/pp-message/pp-message.directive.js (hunk 2 above)
Comment (triitvn) [inline src/directives/pp-message/pp-message.directive.js]: Removed

**Pull request merged**